src/EventLoop/Driver/EventDriver.php

```diff
@@ -210,7 +210,7 @@ protected function activate(array $callbacks): void
             }
 
             if ($callback instanceof TimerCallback) {
-                $interval = \max(0, $callback->expiration - $now);
+                $interval = \min(\max(0, $callback->expiration - $now), \PHP_INT_MAX / 2);
```

Comment:
`PHP_INT_MAX / 2` isn't a great solution, and will likely break in 2038 depending on the internal timer used?